src/batou_ext/nix.py

```diff
@@ -102,37 +102,36 @@ class UserEnv(batou.component.Component):
                     "yarn",
                     "zip",
                 ],
-                channel="<nixos-channel-base-url-to-be-put-in>/nixexprs.tar.xz"
                 shellInit="# additional shell init")
 
     A list of available channels can be found at e.g.
     https://nixos.org/channels/
     """
 
     namevar = "profile_name"
-    channel = batou.component.Attribute(str)
     shellInit = ""
     packages = ()
     let_extra = ""
+    channel = None
 
     def configure(self):
-        self.checksum = hashlib.sha256()
+        if self.channel:
+            batou.output.step(
+                "WARNING",
+                "using `channel` with UserEnv is deprecated, please remove it from your deployment code.",
+                yellow=True,
+            )
+
         template = pkg_resources.resource_string(
             __name__, "resources/userenv.nix"
         ).decode("UTF-8")
         self.profile = self.expand(template).encode("UTF-8")
-        self.checksum.update(self.profile)
-        self.nix_env_name = self.expand(
-            "{{component.profile_name}}-1.{{component.checksum.hexdigest()}}"
-        )
+
         self += batou.lib.file.File(
-            self.profile_name + ".nix", content=self.profile
-        )
-        self += Package(self.nix_env_name, file=self._.path)
-        self.user_profile_path = os.path.expanduser(
-            "~/.nix-profile/etc/profile.d/{}.sh".format(self.profile_name)
+            f"/etc/local/nixos/profile-{self.profile_name}.nix",
+            content=self.profile,
         )
-        self.user_bin_path = os.path.expanduser("~/.nix-profile/bin")
```

Comment:
That's gone why? The new path is somewhere in /etc/ … I'm sure this is used somewhere.

Reply:
Across all projects that I have access to I did not find a single use of this attribute.
Keeping this around is also somewhat redundant since the packages' binaries are automatically added to the user's path.
Additionally, this attribute relies on an implementation detail in nix, the path could change without us noticing and lead to breakage. A deprecation warning similar to the channel attribute might be viable, with a complete removal in a future release of batou_ext.